podman 3.2 fails to build from source (FTBFS) on all architectures that are not amd64, arm64

/kind bug

Please consider the buildlogs at https://buildd.debian.org/status/package.php?p=libpod&suite=experimental

Right now, amd64 and arm64 are the only architectures that podman can build on. On all other architectures, the build fails with:

```
# github.com/containers/podman/pkg/machine
src/github.com/containers/podman/pkg/machine/fcos.go:26:15: undefined: getFCOSDownload
```

The reason is that the function getFCOSDownload is implemented only for these two archs at:
- https://github.com/containers/podman/blob/master/pkg/machine/fcos_amd64.go
- https://github.com/containers/podman/blob/master/pkg/machine/fcos_arm64.go

Please provide either a buildflag to disable this library/functionality or a generic implementation that allows compilation on other architectures.

--- a/pkg/machine/config.go
+++ b/pkg/machine/config.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/connection.go
+++ b/pkg/machine/connection.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/fcos.go
+++ b/pkg/machine/fcos.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/ignition.go
+++ b/pkg/machine/ignition.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/ignition_schema.go
+++ b/pkg/machine/ignition_schema.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 /*

--- a/pkg/machine/keys.go
+++ b/pkg/machine/keys.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/libvirt/config.go
+++ b/pkg/machine/libvirt/config.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package libvirt
 
 type MachineVM struct {

--- a/pkg/machine/libvirt/machine.go
+++ b/pkg/machine/libvirt/machine.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package libvirt
 
 import "github.com/containers/podman/v3/pkg/machine"

--- a/pkg/machine/libvirt/machine_unsupported.go
+++ b/pkg/machine/libvirt/machine_unsupported.go
@@ -1,0 +1,3 @@
+// +build !amd64 amd64,windows
+
+package libvirt

--- a/pkg/machine/machine_unsupported.go
+++ b/pkg/machine/machine_unsupported.go
@@ -1,0 +1,3 @@
+// +build !amd64 amd64,windows
+
+package machine

--- a/pkg/machine/pull.go
+++ b/pkg/machine/pull.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package machine
 
 import (

--- a/pkg/machine/qemu/config.go
+++ b/pkg/machine/qemu/config.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package qemu
 
 import "time"

--- a/pkg/machine/qemu/machine.go
+++ b/pkg/machine/qemu/machine.go
@@ -1,3 +1,5 @@
+// +build amd64,linux arm64,linux amd64,darwin arm64,darwin
+
 package qemu
 
 import (

--- a/pkg/machine/qemu/machine_unsupported.go
+++ b/pkg/machine/qemu/machine_unsupported.go
@@ -1,0 +1,3 @@
+// +build !amd64 amd64,windows
+
+package qemu

